frontend/src/__tests__/cypress/cypress/utils/mockUsers.ts

```diff
@@ -108,38 +108,86 @@ const setUserConfig = (userConfig: UserConfig = {}, isAllowed = true) => {
     req.reply(
       mockSelfSubjectAccessReview({
-        verb,
-        group,
-        resource,
-        namespace,
-        allowed:
-          // cluster admin can do everything
-          isClusterAdmin
-            ? true
-            : // self provisioner capabilities grant access to project creation
-            resource === 'projectrequests'
-            ? isSelfProvisioner
-            : // project admins
-            PROJECT_ADMIN_RESOURCES.includes(resource) && EDIT_VERBS.includes(verb)
-            ? isProjectAdmin
-            : isProductAdmin
-            ? // product admins are getting direct access to resources in the deployment namespace (but importantly, not other projects)
-              namespace === 'opendatahub'
-              ? true
-              : // product admins will be limited to listing resources
-                !EDIT_VERBS.includes(verb)
-            : // everyone else can perform any action within
-              !!namespace,
+        ...resourceAttributes,
+        allowed: evaluateAccess(resourceAttributes, {
+          isClusterAdmin,
+          isProductAdmin,
+          isProjectAdmin,
+          isSelfProvisioner,
+        }),
       }),
     );
   });
 
+  const evaluateAccess = (
+    resourceAttributes: {
+      verb: string;
+      group?: string;
+      resource: string;
+      namespace?: string;
+    },
+    config: UserConfig,
+  ): boolean => {
+    const { verb, resource, namespace } = resourceAttributes;
+    const {
+      isClusterAdmin: clusterAdmin,
+      isProductAdmin: productAdmin,
+      isProjectAdmin: projectAdmin,
+      isSelfProvisioner: selfProvisioner,
+    } = config;
+
+    const PROJECT_ADMIN_RESOURCES = ['projects', 'rolebindings'];
+    const EDIT_VERBS = ['*', 'create', 'delete', 'deletecollection', 'update', 'patch'];
+
+    if (clusterAdmin) {
+      Cypress.log({ message: 'Cluster admin can do everything' });
+      return true;
+    }
+
+    if (resource === 'projectrequests' && selfProvisioner) {
+      Cypress.log({ message: 'Self provisioner capabilities grant access to project creation' });
+      return true;
+    }
+
+    if (resource === 'rolebindings' && !projectAdmin) {
+      Cypress.log({ message: 'Users cannot access permissions tab' });
+      return false;
+    }
+
+    if (PROJECT_ADMIN_RESOURCES.includes(resource) && EDIT_VERBS.includes(verb)) {
+      if (projectAdmin) {
+        Cypress.log({ message: 'Project admins allowed edit on project resources' });
+        return true;
+      }
+      Cypress.log({ message: 'Project users not allowed to edit project resources' });
+      return false;
+    }
+
+    if (productAdmin && namespace === 'opendatahub') {
+      Cypress.log({
+        message:
+          'Product admins are getting direct access to resources in the deployment namespace (but importantly, not other projects)',
+      });
+      return true;
+    }
+
+    if (productAdmin && !EDIT_VERBS.includes(verb)) {
```

Comment:
Don't repeat if-statement checks -- just makes it harder to read... 

```
if (productAdmin) {
  if (namespace...
  if (EDIT_VERBS...
}
```
The goal here is to make this easier to read, and if you simplify it as you step into specific groups it'll be easier to grok even if you've never seen it before "if product admin" -- well no, don't care about those -- then you'd know to skip that block.